client/pages/profile.jsx

```diff
@@ -0,0 +1,312 @@
+import { Button, Col, Input, Row, CardGroup, Card, CardText } from "reactstrap";
+import { Head } from "../components";
+import React, { useState, useEffect } from "react";
+import Link from "next/link";
+import "../public/styles/profile.scss";
+import Select from "react-select";
+import countryData from "../utils/countries";
+import "../public/styles/profileProjects.scss";
+import "../../api/public/schema/projectSchema";
+import { getUser, getModelsByID, getModels } from "../utils/apiWrapper";
+
+export default function Profile() {
+  const [user, setUser] = useState("");
+  const [createdProjects, setCreatedProjects] = useState([]);
+  const [followedProjects, setFollowedProjects] = useState([]);
+  const [projectParsed, setProjectParsed] = useState(false);
+  const [hasUser, setHasUser] = useState(false);
+  const DESCRIPTION_LENGTH = 150;
+  const [country, setCountry] = useState("");
+  const [projects, setProjects] = useState([]);
+
+  useEffect(() => {
+    const populateAllProjects = async () => {
+      const response = await getModels(null, true);
+      setProjects(response.data);
+    };
+    setProjects(projects);
+    populateAllProjects();
+    const getUserInfo = async () => {
+      const response = await getUser();
+      const resp = await response.json();
+      // const resJson = await response.json();
+      // setUser(resJson.data);
+      setUser(resp);
+      setHasUser(true);
+    };
+    getUserInfo();
+  }, []);
+
+  useEffect(() => {
+    async function getProjects() {
+      if (hasUser && !projectParsed && user) {
+        let myprojects = user.data.createdProjects;
+        if (typeof myprojects[0] === "string") {
+          for (let i = 0; i < myprojects.length; i++) {
+            myprojects[i] = await getModelsByID(myprojects[i]);
```

Comment:
We should make an endpoint that returns the users projects in one go. You don't need to do this now, I'll do it later.